### ISSUE TITLE
add include_directories() to template regardless of whether or not depend specified

### ISSUE THREAD

**Issue by jwhendy** [ros-infrastructure/catkin_pkg#167]
Greetings,

**tl;dr:** I'd like to suggest adding the following to `CMakeLists.txt` when `catkin_create_pkg` is called, regardless of whether or not the user specifies dependencies (perhaps as a commented out section?).
```
include_directories(
  ${catkin_INCLUDE_DIRS}
)
```
---

I'm going through [*A Gentle Introduction to ROS*](https://cse.sc.edu/~jokane/agitr/) and was hung up for quite some time on why the first `Hello, world` exercise wouldn't build, failing to find `ros/ros.h` from my include statement. Eventually I stumbled on [the solution](http://answers.ros.org/question/237494/fatal-error-rosrosh-no-such-file-or-directory/), which was to add this line to `CMakeLists.txt`:

```
include_directories(
  ${catkin_INCLUDE_DIRS}
)
```

I've also been through the ROS beginner tutorials in which one also creates a package, and never ran into this issue. I was determined to figure out why, and after much hunting, the only difference is whether or not `catkin_create_pkg` is called with a depend or not.

[ROS wiki](http://wiki.ros.org/ROS/Tutorials/CreatingPackage#ROS.2BAC8-Tutorials.2BAC8-catkin.2BAC8-CreatingPackage.Creating_a_catkin_Package)
```
$ catkin_create_pkg beginner_tutorials std_msgs rospy roscpp
```

*A Gentle Introduction to ROS* (Section 3.1, pg 39)
```
$ catkin_create_pkg agitr
```

I think it would be awesome (if nothing else, for beginners!) to have the `include_directories()` section included just as other template-style (but commented) sections already exist in `CMakeLists.txt`. Here's my reproducible example:

```
$ cd ~/catkin_ws/src
$ catkin_create_pkg beginner_tutorials roscpp
$ catkin_create_pkg beginner_tutorials2

### name-only differences omitted
$ diff --suppress-common-lines beginner_tutorials/CMakeLists.txt beginner_tutorials2/CMakeLists.txt 
10,12c10
< find_package(catkin REQUIRED COMPONENTS
<   roscpp
< )
---
> find_package(catkin REQUIRED)
105,106c103,104
< #  LIBRARIES beginner_tutorials
< #  CATKIN_DEPENDS roscpp
---
> #  LIBRARIES beginner_tutorials2
> #  CATKIN_DEPENDS other_catkin_pkg
117,119d114
< include_directories(
<   ${catkin_INCLUDE_DIRS}
< )
```

Clearly the issue is ultimately on the user (me), but `catkin_create_pkg` already creates *so many sections* in the template that I assumed the possible causes for failure were somewhere in the file. Now that I understand, it doesn't make sense to me that something so basic to building a package with just `ros.h` included *wouldn't* be included in available template options (while many other unnecessary things are).

Thanks for your consideration!

**Comment by dirk-thomas** [comment]
Adding a commented block like this to the template:

```
include_directories(
  ${catkin_INCLUDE_DIRS}
)
```
in the case where it isn't being inserted already would certainly be helpful.

Please consider creating a pull request to add this functionality.

**Comment by jwhendy** [comment]
PR created. First time doing this on `github` and maybe 3rd time ever, so take it easy on me :) (In particular, I wasn't sure what branch to submit to).

I didn't know what to do with the `make test` results... it seems it's checking for some hard-coded result, and thus my output doesn't match what it wants:
```
Traceback (most recent call last):
  File "catkin_pkg/test/test_templates.py", line 108, in test_create_include_macro
    self.assertEqual('# include_directories(include)', statement)
AssertionError: '# include_directories(include)' != '#  include\n#  ${catkin_INCLUDE_DIRS}'
```

I didn't take the time to dig into trying to understand or fix this.

**Comment by dirk-thomas** [comment]
Addressed by #168.
